src/containers/barray.mli

```diff
@@ -46,7 +46,7 @@ val append : 'a array -> 'a array -> 'a array
 val concat : 'a array list -> 'a array
 (** Same as [Array.append], but concatenates a list of arrays. *)
 
-val sub : 'a array -> int -> int -> 'a array
+val slice : 'a array -> int -> int -> 'a array
```

Comment:
If you're going to call this method slice, I think it should honor the same semantics as JS. See the immutable.re implementaiton: https://github.com/facebookincubator/immutable-re/blob/master/src/Immutable.rei#L2098